pkg/runtime/framework/plugins/jobset/jobset_test.go

```diff
@@ -325,3 +330,109 @@ func TestJobSet(t *testing.T) {
+			},
+		},
+		// assert that we get a error  here
+		"dataset initializer job exists but container missing": {
+			runtimeJobTemplate: func() client.Object {
+				js := utiltesting.MakeJobSetWrapper(metav1.NamespaceDefault, "test")
+				for i, rJob := range js.Spec.ReplicatedJobs {
+					if rJob.Name == constants.DatasetInitializer {
+						js.Spec.ReplicatedJobs[i].Template.Spec.Template.Spec.Containers = []v1.Container{}
+					}
+				}
+				return js.Obj()
+			}(),
+			newObj: utiltesting.MakeTrainJobWrapper("default", "test").
+				Initializer(&trainer.Initializer{
+					Dataset: &trainer.DatasetInitializer{},
+				}).Obj(),
+			wantError: field.ErrorList{
+				field.Invalid(field.NewPath("spec").Child("runtimeRef"),
+					utiltesting.MakeTrainJobWrapper("default", "test").Obj().Spec.RuntimeRef,
+					fmt.Sprintf("must have container with name - %s in the %s job", constants.DatasetInitializer, constants.DatasetInitializer)),
+			},
+		},
+		"no model intializer job": {
+			runtimeJobTemplate: utiltesting.MakeJobSetWrapper(metav1.NamespaceDefault, "test"),
+			newObj: utiltesting.MakeTrainJobWrapper(metav1.NamespaceDefault, "test").
+				Initializer(&trainer.Initializer{Model: nil}).
+				Obj(),
+			wantError: field.ErrorList{
+				field.Invalid(field.NewPath("spec").Child("runtimeRef"),
+					utiltesting.MakeTrainJobWrapper("default", "test").Obj().Spec.RuntimeRef,
+					fmt.Sprintf("must have %s job when trainJob is configured with input modelConfig", constants.ModelInitializer)),
+			},
+		},
+		"model intializer job exists but container missing ": {
+			runtimeJobTemplate: func() client.Object {
+				js := utiltesting.MakeJobSetWrapper(metav1.NamespaceDefault, "test")
+				for i, rjob := range js.Spec.ReplicatedJobs {
+					if rjob.Name == constants.ModelInitializer {
+						js.Spec.ReplicatedJobs[i].Template.Spec.Template.Spec.Containers = []v1.Container{}
+					}
+				}
+				return js.Obj()
+			}(),
+			runtimeInfo: runtime.NewInfo(),
+			newObj: utiltesting.MakeTrainJobWrapper("default", "test").
+				Initializer(&trainer.Initializer{
+					Model: &trainer.ModelInitializer{},
+				}).Obj(),
+			wantError: field.ErrorList{
+				field.Invalid(field.NewPath("spec").Child("runtimeRef"),
+					utiltesting.MakeTrainJobWrapper("default", "test").Obj().Spec.RuntimeRef,
+					fmt.Sprintf("must have container with name - %s in the %s job", constants.ModelInitializer, constants.ModelInitializer)),
+			},
+		},
+	}
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			_, ctx := ktesting.NewTestContext(t)
+			var cancel func()
+			ctx, cancel = context.WithCancel(ctx)
+			t.Cleanup(cancel)
+			cli := utiltesting.NewClientBuilder().Build()
+			p, err := New(ctx, cli, nil)
+			if err != nil {
+				t.Fatalf("Failed to initialize JobSet plugin: %v", err)
+			}
+			warnings, errs := p.(framework.CustomValidationPlugin).Validate(tc.runtimeJobTemplate, nil, tc.oldObj, tc.newObj)
```

Comment:
Upstream has update the Validate functions, you can rebase your commit to `main` and remove fist params here.